v2/pkg/catalog/loader/loader.go

```diff
@@ -376,3 +379,49 @@ func workflowContainsProtocol(workflow []*workflows.WorkflowTemplate) bool {
 	}
 	return false
 }
+
+func (store *Store) GetUncoverTargetsFromMetadata(delay, limit int) chan string {
+	templatePaths := store.config.Catalog.GetTemplatesPath(store.finalTemplates)
```

Comment:
this is redundant and can be removed, see previus suggestion. Same for the templates.Parse call below. We do need need to parse the templates again as `store.Load()` does it already